ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/visitor/core/services/collectors/classes/JavaClassIdentityCollector.java

```diff
@@ -0,0 +1,127 @@
+package edu.usb.argos.ASTProcessor.visitor.core.services.collectors.classes;
+
+import edu.usb.argos.ASTProcessor.antlr.JavaParser;
+import edu.usb.argos.ASTProcessor.visitor.core.entities.classes.AnnotationInfo;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.collectors.classes.IClassIdentityCollector;
+import edu.usb.argos.ASTProcessor.visitor.shared.validation.ContextValidator;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.CommonTokenStream;
+
+import java.util.*;
+
+public class JavaClassIdentityCollector implements IClassIdentityCollector<ParserRuleContext> {
+    private final CommonTokenStream tokenStream;
+
+    public JavaClassIdentityCollector(CommonTokenStream tokenStream) {
+        this.tokenStream = tokenStream;
+    }
+
+    @Override
+    public String getClassName(ParserRuleContext ctx) {
+        return ContextValidator.validateAndExecute(
+                ctx,
+                JavaParser.ClassDeclarationContext.class,
+                classCtx -> classCtx.identifier().getText(),
+                ""
+        );
+    }
+
+    @Override
+    public String getPackageName(ParserRuleContext ctx) {
```

Comment:
some classes could not have a package, so maybe you can use the Optional object instead of just returning an empty package name, or maybe you think it's okay that way.

Reply:
Done